src/main/java/com/commercetools/sync/shoppinglists/utils/LineItemUpdateActionUtils.java

```diff
@@ -0,0 +1,257 @@
+        }
+
+        for (int i = indexOfFirstDifference; i < newlineItems.size(); i++) {
+            updateActions.add(AddLineItemWithSku.of(newlineItems.get(i)));
+        }
+
+        return updateActions;
+    }
+
+    private static boolean hasIdenticalAddedAtValues(
+        @Nonnull final LineItem oldLineItem,
+        @Nonnull final LineItemDraft newLineItem) {
+
+        if (newLineItem.getAddedAt() == null) {
+            return true; // omit, if not set in draft.
+        }
+
+        return oldLineItem.getAddedAt().equals(newLineItem.getAddedAt());
+    }
+
+    /**
+     * Compares all the fields of a {@link LineItem} and a {@link LineItemDraft} and returns a list of
+     * {@link UpdateAction}&lt;{@link ShoppingList}&gt; as a result. If both the {@link LineItem} and
+     * the {@link LineItemDraft} have identical fields, then no update action is needed and hence an empty {@link List}
+     * is returned.
+     *
+     * @param oldShoppingList shopping list resource, whose line item should be updated.
+     * @param newShoppingList new shopping list draft, which contains the line item to update.
+     * @param oldLineItem     the line item which should be updated.
+     * @param newLineItem     the line item draft where we get the new fields (i.e. quantity, custom fields and types).
+     * @param syncOptions     responsible for supplying the sync options to the sync utility method. It is used for
+     *                        triggering the error callback within the utility, in case of errors.
+     * @return A list with the update actions or an empty list if the line item fields are identical.
+     */
+    @Nonnull
+    public static List<UpdateAction<ShoppingList>> buildLineItemUpdateActions(
+        @Nonnull final ShoppingList oldShoppingList,
+        @Nonnull final ShoppingListDraft newShoppingList,
+        @Nonnull final LineItem oldLineItem,
+        @Nonnull final LineItemDraft newLineItem,
+        @Nonnull final ShoppingListSyncOptions syncOptions) {
+
+        final List<UpdateAction<ShoppingList>> updateActions = filterEmptyOptionals(
+            buildChangeLineItemQuantityUpdateAction(oldLineItem, newLineItem)
+        );
+
+        updateActions.addAll(
+            buildLineItemCustomUpdateActions(oldShoppingList, newShoppingList, oldLineItem, newLineItem, syncOptions));
+
+        return updateActions;
+    }
+
+    /**
+     * Compares the {@code quantity} values of a {@link LineItem} and a {@link LineItemDraft}
+     * and returns an {@link Optional} of update action, which would contain the {@code "changeLineItemQuantity"}
+     * {@link UpdateAction}. If both {@link LineItem} and {@link LineItemDraft} have the same
+     * {@code quantity} values, then no update action is needed and empty optional will be returned.
+     *
+     * <p>Note: If {@code quantity} from the {@code newLineItem} is {@code null} or {@code 0}, the new {@code quantity}
```

Comment:
Why not set it to zero? Then it will be removed from the list.

Reply:
0 means removing the line item. https://docs.commercetools.com/api/projects/shoppingLists#change-lineitem-quantity

Reply:
Default is 1, so if not set it should be the default in sync context, if you want to remove don't put it to the draft.

Reply:
True that it should not be there. But if it's there anyway with zero, then just update with zero. Don't put default there. It's not expected and additionally we have unnecessary data.

Reply:
fefcd0d